core/src/impl/Kokkos_Core.cpp

```diff
@@ -284,6 +284,8 @@ void finalize_internal( const bool all_spaces = false )
       Kokkos::Cuda::impl_finalize();
 #endif
   }
+#else
+  (void)all_spaces;
```

Comment:
Casting to void sometimes helps, but not always.

Reply:
If there's a better way to do it, I'm happy to change it.

Reply:
@jmgate I wouldn't worry about it unless it actually generates a warning.  The only guaranteed way is to `#ifdef` / `#else` / `#endif` the entire function definition, but that's ugly and could lead to bugs if people forget to implement both sides.

Reply:
Right, I've done that before as well.  I'm just not a fan of `#ifdef`s ☹️

Reply:
I'm with you there :-)